cpp/src/arrow/io/hdfs.cc

```diff
@@ -274,12 +288,20 @@ class HdfsOutputStream::HdfsOutputStreamImpl : public HdfsAnyFileImpl {
     return Status::OK();
   }
 
-  Status Write(const void* buffer, int64_t nbytes, int64_t* bytes_written) {
+  Status Write(const uint8_t* buffer, int64_t nbytes, int64_t* bytes_written) {
```

Comment:
Minor nitpick but why bother with `bytes_written` anymore?  It's discarded when going through the standard FS interfaces and with this change I don't see how it could ever be different from `nbytes` (else you'd never escape the while loop).

Reply:
Good point, it doesn't seem useful indeed. Will remove.